src/internal/pjsdb/queue_crud.go

```diff
@@ -62,3 +64,43 @@ func DequeueAndProcess(ctx context.Context, tx *pachsql.Tx, programHash []byte)
 	}
 	return resp, nil
 }
+
+type GetQueueResponse struct {
+	Size        int64
+	Program     string
+	ProgramHash []byte
+}
+
+type QueueRecord struct {
+	Size    int64  `db:"queue_size"`
+	Program []byte `db:"program"`
+}
+
+func GetQueue(ctx context.Context, tx *pachsql.Tx, queueId []byte) (GetQueueResponse, error) {
+	ctx = pctx.Child(ctx, "getQueue")
+	record := QueueRecord{}
+	err := sqlx.GetContext(ctx, tx, &record, `
+		SELECT COUNT(*) AS queue_size, program
```

Comment:
I think you should merge this query in the queues iterator:
```
SELECT 
				queues.program_hash AS "id", 
				array_agg(j.id ORDER BY j.id ASC) AS "jobs",
				array_agg(j.program ORDER BY j.id ASC) AS "programs",
				count(j.id) AS "size"
			  FROM 
				queues 
			  JOIN 
				pjs.jobs j 
				ON j.program_hash = queues.program_hash
              WHERE
                j.processing IS NULL
				AND j.done IS NULL
				AND j.queued IS NOT NULL
			  GROUP BY 
				queues.program_hash`
```
with this query that gets the size. This way, there won't be duplicate queries. We can also ensure that we'd be using the same query to get 1 queue that we use to get all queues.